swap/src/bin/swap_cli.rs

```diff
@@ -107,6 +107,14 @@ async fn main() -> Result<()> {
             alice_peer_id,
             alice_addr,
         } => {
+            if receive_monero_address.network != monero_network {
+                bail!(
+                    "Given monero address is on network {:?}, expected address on network {:?}",
```

Comment:
Can we use Display instead?

Reply:
The `monero::Network` does not implement `Display` - but it is an `enum`, so it looks like this:

```
Error: Given monero address is on network Mainnet, expected address on network Stagenet
```


I did not see the necessity to add a custom `Display` impl, given that it loops good anyway. 
Should we upstream a display impl/derive ...?

Reply:
If that is what the `Debug` print looks like then it is fine :)

Reply:
> If that is what the `Debug` print looks like then it is fine :)

It is - I copied it after running it with a mainnet address :)